### PR TITLE
keadm beta config images: support remote runtime

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/beta/config.go [hunk 1]
```diff
@@ -169,6 +172,12 @@ func AddImagesCommonConfigFlags(cmd *cobra.Command, cfg *Configuration) {
 	)
 	cmd.Flags().StringVar(&cfg.Part, "part", cfg.Part,
 		"Use this key to set which part keadm will install: cloud part or edge part. If not set, keadm will list/pull all images used by both cloud part and edge part.")
+
+	cmd.Flags().StringVar(&cfg.RuntimeType, cmdcommon.RuntimeType, cfg.RuntimeType,
+		"Container runtime type, default is docker")
+
+	cmd.Flags().StringVar(&cfg.RemoteRuntimeEndpoint, cmdcommon.RemoteRuntimeEndpoint, cfg.RemoteRuntimeEndpoint,
+		"KubeEdge Edge Node RemoteRuntimeEndpoint string")
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/beta/config.go:180]: s/KubeEdge Edge Node RemoteRuntimeEndpoint string/remoteRuntimeEndpoint is the endpoint of remote runtime service in edge node/g

Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/beta/config.go:180]: s/KubeEdge Edge Node RemoteRuntimeEndpoint string/remoteRuntimeEndpoint is the endpoint of remote runtime service in edge node/g

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go:180]: fixed to `The endpoint of remote runtime service in edge node`

File: keadm/cmd/keadm/app/cmd/beta/config.go [hunk 2]
```diff
@@ -227,8 +236,62 @@ func GetPauseImage() string {
...
+	if err != nil {
+		return err
+	}
+	if status == nil || status.Id == "" {
+		if _, err := imageService.PullImage(imageSpec, nil, nil); err != nil {
+			return err
+		}
+	}
+
+	return nil
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/beta/config.go:281]: if image not exist, we should return err

File: keadm/cmd/keadm/app/cmd/beta/config.go [hunk 3]
```diff
@@ -227,8 +236,62 @@ func GetPauseImage() string {
...
+}
+
+type DockerRuntime struct {
+}
+
+func (runtime *DockerRuntime) PullImage(image string) error {
+	cli, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv)
+	if err != nil {
+		return fmt.Errorf("init docker dockerclient failed: %v", err)
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/beta/config.go:290]: s/init docker dockerclient failed/init docker client failed/g

File: keadm/cmd/keadm/app/cmd/beta/config.go (hunk 2 above)
Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go:281]: line 275: this `status == nil || status.Id == ""` block represent the image not exist
line 281 represent the image exist or pull image successfully.

File: keadm/cmd/keadm/app/cmd/beta/config.go (hunk 3 above)
Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go:290]: yes

File: keadm/cmd/keadm/app/cmd/beta/config.go (hunk 2 above)
Comment (wackxu) [reply keadm/cmd/keadm/app/cmd/beta/config.go:281]: got it

File: keadm/cmd/keadm/app/cmd/beta/config.go (hunk 3 above)
Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go]: fixed

File: keadm/cmd/keadm/app/cmd/beta/config.go [hunk 4]
```diff
@@ -227,8 +236,62 @@ func GetPauseImage() string {
...
+
+type DockerRuntime struct {
+}
+
+func (runtime *DockerRuntime) PullImage(image string) error {
+	cli, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv)
+	if err != nil {
+		return fmt.Errorf("init docker client failed: %v", err)
+	}
+
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/beta/config.go:292]: I think  we should call docker `NegotiateAPIVersion` to solve docker client and server API version mismatch problem. I have encountered this problem ever

**Review by wackxu — APPROVED** [review]
lgtm

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @zc2638

**Pull request merged**